core/src/test/java/org/infinispan/persistence/file/SingleFileStoreTest.java

```diff
@@ -109,4 +111,21 @@ public void testSegmentsRemovedAndAdded() {
 
       assertTrue(store.contains(key));
    }
+
+   public void testStopDuringClear() {
```

Comment:
I am not sure this test provides much as the PersistenceManagerImpl has guarantees to not call stop or start during any other store operation, but no reason for this stop integration.

Reply:
True. I started doing only the truncate(4) bit thinking about crashes, and then I got a bit carried away.